TTS/tts/models/vits.py

```diff
@@ -1318,22 +1370,38 @@ def format_batch_on_device(self, batch):
         """Compute spectrograms on the device."""
         ac = self.config.audio
 
+        if self.args.TTS_part_sample_rate:
+            wav = self.audio_resampler(batch["waveform"])
+        else:
+            wav = batch["waveform"]
+
         # compute spectrograms
-        batch["spec"] = wav_to_spec(batch["waveform"], ac.fft_size, ac.hop_length, ac.win_length, center=False)
+        batch["spec"] = wav_to_spec(wav, ac.fft_size, ac.hop_length, ac.win_length, center=False)
+
+        if self.args.TTS_part_sample_rate:
+            # recompute spec with high sampling rate to the loss
+            spec_mel = wav_to_spec(batch["waveform"], ac.fft_size, ac.hop_length, ac.win_length, center=False)
+        else:
+            spec_mel = batch["spec"]
+
         batch["mel"] = spec_to_mel(
-            spec=batch["spec"],
+            spec=spec_mel,
             n_fft=ac.fft_size,
             num_mels=ac.num_mels,
             sample_rate=ac.sample_rate,
             fmin=ac.mel_fmin,
             fmax=ac.mel_fmax,
         )
-        assert batch["spec"].shape[2] == batch["mel"].shape[2], f"{batch['spec'].shape[2]}, {batch['mel'].shape[2]}"
+
+        if not self.args.TTS_part_sample_rate:
+            assert batch["spec"].shape[2] == batch["mel"].shape[2], f"{batch['spec'].shape[2]}, {batch['mel'].shape[2]}"
```

Comment:
can you also implement the corresponding assert when upsampling is used?

Reply:
Done